### PR TITLE
WASM: Use passes array_op, implied_do_loop, select_case

### PR DESCRIPTION
**Pull request by ubaidsk**
fixes https://github.com/lfortran/lfortran/issues/1366#issuecomment-1489742931

Also towards https://github.com/lcompilers/lpython/issues/1485

### CODE REVIEW

File: src/libasr/codegen/asr_to_wasm.cpp [hunk 1]
```diff
@@ -2899,8 +2902,11 @@ Result<Vec<uint8_t>> asr_to_wasm_bytes_stream(ASR::TranslationUnit_t &asr,
 
     LCompilers::PassOptions pass_options;
     pass_array_by_data(al, asr, pass_options);
+    pass_replace_array_op(al, asr, pass_options);
+    pass_replace_implied_do_loops(al, asr, pass_options);
     pass_replace_print_arr(al, asr, pass_options);
     pass_replace_do_loops(al, asr, pass_options);
+    pass_replace_select_case(al, asr, pass_options);
```
Comment (czgdp1807) [inline src/libasr/codegen/asr_to_wasm.cpp:2909]: We should use `pass_manager.h`. See Julia backend on how we use pass_manager to call limited number of passes.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: src/libasr/codegen/asr_to_wasm.cpp (hunk 1 above)
Comment (Shaikh-Ubaid) [reply src/libasr/codegen/asr_to_wasm.cpp:2909]: I looked into the `asr_to_julia` backends in `lpython` and `lfortran` and also the `emit_julia()` in `src/bin/lfortran.cpp`. There does not seem a `pass_manager` used. Do you mean to use the function `pass_manager.parse_pass_arg()`?

Comment (czgdp1807) [reply src/libasr/codegen/asr_to_wasm.cpp:2909]: https://github.com/lfortran/lfortran/pull/1500#discussion_r1161299168 should help.

Comment (Shaikh-Ubaid) [reply src/libasr/codegen/asr_to_wasm.cpp]: Done. Thank you for the guidance and support!

**Comment by Shaikh-Ubaid** [comment]
This is ready. Please review and share feedback.

**Review by czgdp1807 — APPROVED** [review]

**Review by certik — APPROVED** [review]

**Pull request merged**